app-backend/db/src/main/resources/migrations/V8__Add_cascading_deletion_to_user_fks.sql

```diff
@@ -2,124 +2,237 @@ BEGIN;
-ADD CONSTRAINT "map_tokens_created_by_fkey" FOREIGN KEY (created_by) REFERENCES users(id) ON DELETE CASCADE,
-ADD CONSTRAINT "map_tokens_modified_by_fkey" FOREIGN KEY (modified_by) REFERENCES users(id) ON DELETE CASCADE;
+ADD CONSTRAINT "map_tokens_created_by_fkey" FOREIGN KEY (created_by) REFERENCES users(id) ON DELETE CASCADE NOT VALID,
+ADD CONSTRAINT "map_tokens_modified_by_fkey" FOREIGN KEY (modified_by) REFERENCES users(id) ON DELETE SET NULL NOT VALID,
+ADD CONSTRAINT "map_tokens_owner_fkey" FOREIGN KEY (owner) REFERENCES users(id) ON DELETE CASCADE NOT VALID;
 
 ALTER TABLE "projects"
+ALTER COLUMN "modified_by" DROP NOT NULL,
 DROP CONSTRAINT "projects_created_by_fkey",
 DROP CONSTRAINT "projects_modified_by_fkey",
 DROP CONSTRAINT "projects_owner_fkey",
-ADD CONSTRAINT "projects_created_by_fkey" FOREIGN KEY (created_by) REFERENCES users(id) ON DELETE CASCADE,
-ADD CONSTRAINT "projects_modified_by_fkey" FOREIGN KEY (modified_by) REFERENCES users(id) ON DELETE CASCADE;
+ADD CONSTRAINT "projects_created_by_fkey" FOREIGN KEY (created_by) REFERENCES users(id) ON DELETE CASCADE NOT VALID,
+ADD CONSTRAINT "projects_modified_by_fkey" FOREIGN KEY (modified_by) REFERENCES users(id) ON DELETE SET NULL NOT VALID,
+ADD CONSTRAINT "projects_owner_fkey" FOREIGN KEY (owner) REFERENCES users(id) ON DELETE CASCADE NOT VALID;
 
 ALTER TABLE "scenes"
+ALTER COLUMN "modified_by" DROP NOT NULL,
 DROP CONSTRAINT "scenes_created_by_fkey",
 DROP CONSTRAINT "scenes_modified_by_fkey",
 DROP CONSTRAINT "scenes_owner_fkey",
-ADD CONSTRAINT "scenes_created_by_fkey" FOREIGN KEY (created_by) REFERENCES users(id) ON DELETE CASCADE,
-ADD CONSTRAINT "scenes_modified_by_fkey" FOREIGN KEY (modified_by) REFERENCES users(id) ON DELETE CASCADE;
+ADD CONSTRAINT "scenes_created_by_fkey" FOREIGN KEY (created_by) REFERENCES users(id) ON DELETE CASCADE NOT VALID,
+ADD CONSTRAINT "scenes_modified_by_fkey" FOREIGN KEY (modified_by) REFERENCES users(id) ON DELETE SET NULL NOT VALID,
+ADD CONSTRAINT "scenes_owner_fkey" FOREIGN KEY (owner) REFERENCES users(id) ON DELETE CASCADE NOT VALID;
 
 ALTER TABLE "shapes"
+ALTER COLUMN "modified_by" DROP NOT NULL,
 DROP CONSTRAINT "shapes_created_by_fkey",
 DROP CONSTRAINT "shapes_modified_by_fkey",
-ADD CONSTRAINT "shapes_created_by_fkey" FOREIGN KEY (created_by) REFERENCES users(id) ON DELETE CASCADE,
-ADD CONSTRAINT "shapes_modified_by_fkey" FOREIGN KEY (modified_by) REFERENCES users(id) ON DELETE CASCADE;
+ADD CONSTRAINT "shapes_created_by_fkey" FOREIGN KEY (created_by) REFERENCES users(id) ON DELETE CASCADE NOT VALID,
+ADD CONSTRAINT "shapes_modified_by_fkey" FOREIGN KEY (modified_by) REFERENCES users(id) ON DELETE SET NULL NOT VALID;
 
 ALTER TABLE "task_actions"
 DROP CONSTRAINT "task_actions_user_id_fkey",
-ADD CONSTRAINT "task_actions_user_id_fkey" FOREIGN KEY (user_id) REFERENCES users(id) ON DELETE CASCADE;
+ADD CONSTRAINT "task_actions_user_id_fkey" FOREIGN KEY (user_id) REFERENCES users(id) ON DELETE SET NULL NOT VALID;
```

Comment:
It looks like `user_id ` field has a `NOT NULL` constraint on it.

Reply:
Good catch